### PR TITLE
Add configuration management service

### PR DESCRIPTION
**Pull request by timraymond**
This sets up a configuration managemet service on a configureable port
(which itself is configureable by the configuration management
service!). It sets up an HTTP handler on this port whereby GETs to /
will deliver the current TOML configuration that Telegraf booted from.
POSTs to / will update that configuration. Telegraf can then be
restarted from the configuration by sending SIGHUP through existing
mechanisms.

This configuration management service is not authenticated in any way!
It is assumed that this service will only be available behind the
firewall.

### CODE REVIEW

File: internal/config/config.go [hunk 1]
```diff
@@ -175,6 +178,9 @@ var header = `# Telegraf configuration
   # Override default hostname, if empty use os.Hostname()
   hostname = ""
 
+	# Set configuration management port
```
Comment (sparrc) [inline internal/config/config.go]: these should be indented by 2 spaces rather than tabs

File: internal/config/config.go [hunk 2]
```diff
@@ -175,6 +178,9 @@ var header = `# Telegraf configuration
...
+	config-port = ""
```
Comment (sparrc) [inline internal/config/config.go]: I believe the test is failing because this should be `config_port`

**Comment by sparrc** [comment]
we have a separate PR for doing configuration management via etcd: https://github.com/influxdata/telegraf/pull/651, would that accomplish the same task?

**Comment by timraymond** [comment]
It _would_, but Managed Hosting isn't quite at the point where they can make use of it at the moment. Configuration management via Etcd would likely be combined with some larger changes on their side. I'm told that will happen eventually, but this will be easier for them to use for the time being.

**Comment by sparrc** [comment]
The service already has the SIGHUP method of telling it to reload it's config, why not just change the config file on disk before sending the SIGHUP? What does having this service running do?

I guess what I'm saying is that it doesn't seem any easier to change the file via a POST vs. just changing it on disk directly.

**Comment by timraymond** [comment]
That seems simpler to me... @kfitzpatrick does that :point_up: work for you?

**Comment by sparrc** [comment]
I'm going to close this unless there is significant opposition. Exposing and changing the config file over HTTP seems like a mess IMHO.

The SIGHUP mechanism exists for the purpose of reloading the config file on disk, I see this PR creating the following problem very easily:

1. admin starts telegraf service
1. admin POSTs new config to endpoint
1. admin SIGHUPs telegraf service so that it reloads the new config
1. admin restarts telegraf service or reboots box (or service dies and restarts itself).
1. telegraf is now running again with the _on disk_ config, and has no memory of the POSTed config.

My main question is: why create this problem? Why not just change the file on disk?

**Pull request closed without merging**